lib/mailserver.js

```diff
@@ -476,6 +470,29 @@ mailServer.getEmailEml = function (id, done) {
 
     var filename = email.id + '.eml'
 
-    done(null, 'message/rfc822', filename, fs.createReadStream(path.join(tempDir, id + '.eml')))
+    done(null, 'message/rfc822', filename, fs.createReadStream(path.join(mailServer.mailDir, id + '.eml')))
+  })
+}
+
+mailServer.loadMailsFromDirectory = function () {
+  var persistencePath = fs.realpathSync(mailServer.mailDir)
+  store.length = 0
+  fs.readdir(persistencePath, function (err, files) {
+    if (err) {
+      logger.error('Error during reading of the mailDir %s', persistencePath)
+    } else {
+      files.forEach(function (file) {
+        var filePath = persistencePath + '/' + file
+        fs.readFile(filePath, 'utf8', function (err, data) {
+          if (err) {
+            logger.error('Error during reading of the file %s', filePath)
+          } else {
+            var mail = JSON.parse(data)
```

Comment:
Thanks for the PR. I've tried your image without much success. This line seems to be the problem. Isn't this supposed to parse the eml file (which is not a JSON)?

Reply:
My bad, I forgot to add some code of my fork.
2 ways to deal with this :

- Persist the state in a json file per eml
- Reload the state from all eml file

Reply:
New commits to reload mails from eml

Reply:
I updated the docker image and charts : https://github.com/maildev/maildev/pull/303#issuecomment-630999150